sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -510,16 +516,24 @@ func (m *networkFlowManager) enrichHostConnections(hostConns *hostConnections, e
 	flowMetrics.HostConnectionsRemoved.Add(float64(prevSize - len(hostConns.connections)))
 }
 
+func deleteEndpoint(hostConns *hostConnections, ep containerEndpoint) {
+
+	delete(hostConns.endpoints, ep)
+	processID := getProcessInfoString(ep.processKey)
+	delete(hostConns.processes, processID)
+}
+
 func (m *networkFlowManager) enrichHostContainerEndpoints(hostConns *hostConnections, enrichedEndpoints map[containerEndpointIndicator]timestamp.MicroTS) {
 	hostConns.mutex.Lock()
 	defer hostConns.mutex.Unlock()
 
 	prevSize := len(hostConns.endpoints)
 	for ep, status := range hostConns.endpoints {
 		m.enrichContainerEndpoint(&ep, status, enrichedEndpoints)
-		if status.used && status.lastSeen != timestamp.InfiniteFuture {
+		if status.used && (status.usedProcess || !features.ProcessesListeningOnPort.Enabled()) && status.lastSeen != timestamp.InfiniteFuture {
```

Comment:
Please, untangle this condition, it's unreadable. Maybe introduce a local `used` variable computed from the status and the feature enabled or such.

Reply:
Done

Reply:
Done